packages/core/entities-config-editor/src/composables/useMonacoEditor.ts

```diff
@@ -0,0 +1,88 @@
+import { cloneDeep } from 'lodash-es'
+import type * as Monaco from 'monaco-editor'
+import { v4 as uuid } from 'uuid'
+import {
+  getLanguageService as getJSONLanguageService,
+  TextDocument,
+  type JSONDocument,
+} from 'vscode-json-languageservice'
+import { computed, shallowRef, unref, type MaybeRef } from 'vue'
+import { type EditorLanguage, type LuaSchema } from '../types'
+import { buildRecordSchema } from '../utils'
+
+declare global {
+  interface Window {
+    require: any
+  }
+}
+
+const jsonLanguageService = getJSONLanguageService({})
+
+export default function useMonacoEditor(
+  monaco: typeof Monaco,
+  language: MaybeRef<EditorLanguage>,
+  documentId: MaybeRef<string | undefined>,
+  luaSchema: MaybeRef<LuaSchema | undefined>,
+) {
+  const uri = computed(() => {
+    const documentName = unref(documentId) || `unsaved_${uuid()}`
+    return monaco.Uri.parse(`kong:///editor/entities/${documentName}.${unref(language)}`)
+  })
+
+  const schema = computed(() => {
+    const _luaSchema = unref(luaSchema)
+    if (!_luaSchema) {
+      return undefined
+    }
+
+    const rootSchema = cloneDeep(buildRecordSchema(_luaSchema, unref(language)))
+    rootSchema.properties = {
+      ...rootSchema.properties,
+    }
+    return rootSchema
+  })
+
+  const model = computed(() => {
+    const _language = unref(language)
+    const existingModel = monaco.editor.getModel(uri.value)
+    if (existingModel) {
+      existingModel.dispose()
+    }
+    return monaco.editor.createModel(_language === 'json' ? '{}' : '', _language, uri.value)
+  })
+
+  const textDocument = computed(() =>{
+    const _language = unref(language)
+    return TextDocument.create(uri.value.toString(), _language, 0, _language === 'json' ? '{}' : '')
+  })
```

Comment:
WDYT about isolating the ternary expression into computed like:

```ts
const langTemplate = computed(() => language.value === 'json' ? '{}' : '')
```

Reply:
Thank you for the suggestion. I'm marking this pull request as Draft for now as I plan to refactor the code a little bit. Let me come back with more improvements.

Reply:
Cool, let me know when it's ready ✉️